app/models/resident.rb

```diff
@@ -1,5 +1,15 @@
 # frozen_string_literal: true
 
+require 'acts_as_list'
+
 class Resident < ActiveRecord::Base
-  validates_uniqueness_of :name, case_sensitive: false
+  belongs_to :user
+
+  validates_uniqueness_of :name, case_sensitive: false, scope: :user_id
+
+  acts_as_list scope: :user
+
+  def next_resident
+    lower_item || user.residents.limit(1).first
+  end
```

Comment:
So I'm guessing this gem is just doing all the work for us here and keeping up with our place in the list as we go? On line 13 we're essentially just saying give me the next resident or give me the first one if nil?

Reply:
The gem doesn't really "keep track" of the position, but if we have an instance of a Resident we can navigate down the list using `lower_item`. This returns the next record in the list relative to the instance you called it on.